### DISCUSSION TITLE
Adding Orchard Core CMS to an existing application that uses ASP.NET Core Identity

### DISCUSSION THREAD

**Discussion by DrewBrasher** [OrchardCMS/OrchardCore#9832] [Q&A]
I have some existing .NET Core applications that were created using the "Individual User Accounts" option which uses ASP.NET Core Identity.  Is there a way to add Orchard Core CMS to this application and have it use the existing authentication system?

**Comment by Skrypt** [comment]
You mean like a single sign on for both apps? I'm pretty sure it's possible but don't know all the details.

**Reply by DrewBrasher** [reply]
What I had in mind was adding the CMS to an existing web project so it would all be one app but after trying to do that with the Visual Studio MVC template I'm not sure that is even possible.  I haven't spent much time trying it yet but the home controller routes were conflicting with the routes in Orchard Core CMS.

**Reply by Skrypt** [reply]
The home route is an easy fix it's documented.

**Comment by ns8482e** [comment]
I guess ASP.NET Core Identity is not supported AS IS, you need to implement your own  `UserStore` and replace following DI added by Users module with your own store

https://github.com/OrchardCMS/OrchardCore/blob/e2a1b00142fa7703de35ed57ab504cf0ff658952/src/OrchardCore.Modules/OrchardCore.Users/Startup.cs#L164-L173

**Reply by DrewBrasher** [reply]
Thanks, I will try that if I am able to add the CMS to an existing MVC site.
